config/okta.spc

```diff
@@ -3,10 +3,29 @@ connection "okta" {
 
   # Get your API token from Okta https://developer.okta.com/docs/guides/create-an-api-token/create-the-token/
   # domain = "https://<your_okta_domain>.okta.com"
+
+  # Okta API token. Can also be set with the OKTA_CLIENT_TOKEN environment variable.
   # token  = "02d0YZgNSJwlNew6lZG-6qGThisisatest-token"
 
   # Or use an Okta application and the client credentials flow for authenticating: https://developer.okta.com/docs/guides/implement-oauth-for-okta-serviceapp/overview/
   # domain      = "https://<your_okta_domain>.okta.com"
+
+  # Okta App client id, used with PrivateKey OAuth auth mode. Can also be set with the OKTA_CLIENT_CLIENTID environment variable.
   # client_id   = "0oa10zpa2bo6tAm9Test"
+
+  # Private key value. Can also be set with the OKTA_CLIENT_PRIVATEKEY environment variable.
   # private_key = "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAK..."
+
+  # The maximum number of attempts (including the initial call) Steampipe will
+  # make for failing API calls. Can also be set with the OKTA_CLIENT_RATE_LIMIT_MAX_RETRIES environment variable.
+  # Defaults to 5 and must be greater than or equal to 1.
+  # max_retries = 5
+
+  # The maximum amount of time to wait on request back off. Can also be set with the OKTA_CLIENT_RATE_LIMIT_MAX_BACKOFF environment variable.
+  # Defaults to 30 and must be greater than or equal to 1.
+  # max_backoff = 30
+
+  # HTTP request time out in seconds. Can also be set with the OKTA_CLIENT_REQUEST_TIMEOUT environment variable.
+  # Defaults to 30 and must be greater than or equal to 1.
+  # request_timeout = 30
```

Comment:
@ParthaI, is there any specific reason you chose the above default values for the new config parameters? Does Okta have any documentation on it?

Reply:
I think we can add this doc link - https://github.com/okta/okta-sdk-golang?tab=readme-ov-file#environment-variables

Reply:
@ParthaI how did you choose the naming convention of the env variables?

Reply:
@misraved, I made an interesting observation while exploring how the Okta SDK generates its configuration from environment variables. The naming conventions were clear upon reviewing the `config.go` file within the Okta SDK itself, available [here](https://github.com/okta/okta-sdk-golang/blob/master/okta/config.go#L33-L70).

Conversely, the [README](https://github.com/okta/okta-sdk-golang?tab=readme-ov-file#environment-variables) document outlines a specific formula for configuring Okta environment variables, which seems to differ from the method described above.

Regarding the default value for `OKTA_CLIENT_RATE_LIMIT_MAX_RETRIES`, I opted to adapt the retry logic implementation similar to that of the AWS plugin, setting the default value to 9, diverging from the documentation [here](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/retries-timeouts/) which states it is 3. As for the `OKTA_CLIENT_RATE_LIMIT_MAX_BACKOFF` default value, I aligned it with the guidance provided [here](https://github.com/okta/okta-sdk-golang?tab=readme-ov-file#okta-client-base-configuration).